tests/framework/data/SortTest.php

```diff
@@ -53,6 +53,35 @@ public function testGetOrders()
         $this->assertEquals(SORT_ASC, $orders['age']);
     }
 
+    public function testGetExpressionOrders()
+    {
+        $sort = new Sort([
+            'attributes' => [
+                'age',
+                'name' => [
+                    'asc' => ['first_name' => SORT_ASC, '[[last_name]] ASC NULLS FIRST'],
+                    'desc' => ['first_name' => SORT_DESC, '[[last_name]] DESC NULLS LAST'],
+                ],
+            ],
+            'params' => [
+                'sort' => 'age,-name'
+            ],
+            'enableMultiSort' => true,
+        ]);
+
+        $orders = $sort->getOrders();
+        $this->assertEquals(3, count($orders));
+        $this->assertEquals(SORT_ASC, $orders['age']);
+        $this->assertEquals(SORT_DESC, $orders['first_name']);
+        $this->assertEquals('[[last_name]] DESC NULLS LAST', $orders[0]);
+        $this->assertEquals(['age', 'first_name', 0], array_keys($orders));
+
+        $sort->enableMultiSort = false;
+        $orders = $sort->getOrders(true);
+        $this->assertEquals(1, count($orders));
+        $this->assertEquals(SORT_ASC, $orders['age']);
+    }
```

Comment:
I would simplify this test: all waht is necessary is check if expression can be passed as sort attribute value.
There is no need to check multisort here as it is already covered.